### PR TITLE
Propagate error messages from 3PID and display them during signup

### PR DESCRIPTION
**Pull request by OneBlue**
Hello ! 

This change corrects a bug where the sign-up flow hangs forever if the 3PID token request fails.

If an error occurs, the error is displayed as a Toast message and the user is taken back to the previous form.

### Pull Request Checklist

* [X] Pull request is based on the develop branch
* [X] Pull request updates [CHANGES.rst](https://github.com/vector-im/riot-android/blob/develop/CHANGES.rst)
* [ ] Pull request includes screenshots or videos if containing UI changes
* [X] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

File: vector/src/main/java/im/vector/RegistrationManager.java [hunk 1]
```diff
@@ -890,15 +890,15 @@ public void onNetworkError(final Exception e) {
 
                         @Override
                         public void onUnexpectedError(Exception e) {
-                            listener.onThreePidRequested(pid);
+                            listener.onThreePidRequestFailed(R.string.account_email_error, e.toString());
```
Comment (bmarty) [inline vector/src/main/java/im/vector/RegistrationManager.java]: Correct code should be:
`listener.onThreePidRequestFailed(R.string.account_phone_number_error, e.getLocalizedMessage());`

File: vector/src/main/java/im/vector/RegistrationManager.java [hunk 2]
```diff
@@ -862,15 +862,15 @@ public void onNetworkError(final Exception e) {
...
                         }
 
                         @Override
                         public void onMatrixError(MatrixError e) {
                             if (TextUtils.equals(MatrixError.THREEPID_IN_USE, e.errcode)) {
-                                listener.onThreePidRequestFailed(R.string.account_email_already_used_error);
+                                listener.onThreePidRequestFailed(R.string.account_email_already_used_error, null);
                             } else {
-                                listener.onThreePidRequested(pid);
+                                listener.onThreePidRequestFailed(R.string.account_email_error, e.error);
```
Comment (bmarty) [inline vector/src/main/java/im/vector/RegistrationManager.java]: Please replace `e.error` to `e.getLocalizedMessage()`

**Comment by OneBlue** [comment]
Thank you for the review @bmarty. 

I updated my branch accordingly.

**Comment by bmarty** [comment]
Hi @OneBlue,

Thanks for the update, looks good to me

Benoît

**Pull request merged**